CodeGeneration/Sources/SyntaxSupport/SyntaxNodeKind.swift

```diff
@@ -367,6 +367,19 @@ public enum SyntaxNodeKind: String, CaseIterable {
     }
   }
 
+  /// For base node types, generates the name of the protocol to which all
+  /// concrete leaf nodes that derive from this base kind should conform.
+  ///
+  /// - Warning: This property can only be accessed for base node kinds; attempting to
+  /// access it for a non-base kind will result in a runtime error.
+  public var leafProtocolType: TypeSyntax {
+    if isBase {
+      return "Leaf\(syntaxType)NodeProtocol"
```

Comment:
Sorry, I should have noted this earlier. Since this protocol needs to be public so that that methods defined within it are visible publicly but since we don’t want to offer any guarantees about supporting it in the future, we should prefix it with an underscore.

```suggestion
      return "_Leaf\(syntaxType)NodeProtocol"
```

Reply:
Done